### DISCUSSION TITLE
Can I use `cache` in a regular API route?

### DISCUSSION THREAD

**Discussion by mastoj** [vercel/next.js#51295] [App Router]
https://nextjs.org/docs/app/building-your-application/data-fetching/caching the documentation here only refers to server side components it seems, will it also inside a regular api route?

**Comment by mogeko** [comment]
I'm not sure if the Cache function of APP Router is also suitable for Page Router, but as an alternative, if you want to **create an API Router in App Router**, you can use the [`route.js` file](https://nextjs.org/docs/app/building-your-application/routing/router-handlers).

**Comment by Andrewjeska** [comment]
Also wondering this

**Comment by funes79** [comment]
Yes, API routes in APP router (for example `app/api/mycategories/route.js`) are also subject to caching. 
For example a GET method is always cached (without request parameters).

So let say you have an API route to return items from a db table. This route will be compiled during build - and if your connection string does not work in build because of missing credentials - then the API will always return empty data. 

But you can configure the route to be always non cached using defining dynamic global variable: 
```
export const dynamic = 'force-dynamic';

export async function GET() {
  const token = getSnowflakeAuthToken();
  const data = await selectTable(....);
  ...
}
```

or if you want to hit the db at most every 20 seconds you define 
```
export const revalidate = 20;

export async function GET() {
  const token = getSnowflakeAuthToken();
  const data = await selectTable(....);
  ...
}
```
So in this case when you call the route first time, it will read the data. Then if you call it within the period of 20s, the call hits the server (if you use middleware.js - you can see the call coming in) - but the function code is never executed, just data is returned from the cache. And after the 20s if you call this api route again, the function executes.

I just did not figure out how to enforce the invalidation of cache on API routes. `revalidatePath` does not seem to work for me (called both from client and server side).

**Reply by programming-with-ia** [reply]
🚀, does export `const revalidate = 20` even do anything on a `POST` in same `route.ts`?
